metricflow/sql/optimizer/rewriting_sub_query_reducer.py

```diff
@@ -579,6 +578,13 @@ def _rewrite_node_with_join(node: SqlSelectStatementNode) -> SqlSelectStatementN
             select_columns=tuple(clauses_to_rewrite.select_columns),
             from_source=from_source,
             from_source_alias=from_source_alias,
+            cte_sources=tuple(
+                SqlCteNode.create(
```

Comment:
Should this use `cte_source.with_new_select`?

Reply:
Good catch. `with_new_select` got added as a convenience method late in the stack.